src/lib/Scenes/Artwork/Components/ContextCard.tsx

```diff
@@ -0,0 +1,238 @@
+      <Button
+        variant={is_followed ? "secondaryOutline" : "primaryBlack"}
+        onPress={() => this.handleFollowShow(show)}
+        size="small"
+        longestText="Following"
+        loading={isSaving}
+      >
+        {is_followed ? "Following" : "Follow"}
+      </Button>
+    )
+  }
+
+  render() {
+    let name = ""
+    let href = ""
+    let meta = null
+    let header = ""
+    let imageUrl = ""
+    let followButton = null
+    let renderContextCard = false
+
+    const { artwork } = this.props
+    const { context } = artwork
+
+    if (context) {
+      const { __typename } = context
+      switch (__typename as any) {
+        case "ArtworkContextAuction":
+          header = "In auction"
+          name = context.name
+          href = context.href
+          meta = context.formattedStartDateTime
+          imageUrl = context.cover_image && context.cover_image.url ? context.cover_image.url : ""
+          renderContextCard = true
+          break
+        case "ArtworkContextFair":
+          const { fair } = artwork
+          header = "In fair"
+          name = fair.name
+          href = fair.href
+          meta = fair.exhibition_period
+          imageUrl = fair.image && fair.image.url ? fair.image.url : ""
+          renderContextCard = true
+          break
+        case "ArtworkContextPartnerShow":
+          // TODO: Replace with ArtworkContextShow when MPv2 supports it
+          const { shows } = artwork
+          const show = shows[0]
+          header = "In show"
+          name = show.name
+          href = show.href
+          meta = show.exhibition_period
+          imageUrl = show.cover_image && show.cover_image.url ? show.cover_image.url : ""
+          followButton = this.followButton(show)
+          renderContextCard = true
+          break
+      }
+    }
+
+    return (
+      <>
+        {renderContextCard && (
+          <Box mb={2}>
+            <Sans size="3t" weight="medium" color="black100">
+              {header}
+            </Sans>
+          </Box>
+        )}
+        {renderContextCard && (
+          <Flex>
+            <TouchableWithoutFeedback onPress={() => this.handleTap(this, href)}>
+              <EntityHeader
+                name={name}
+                href={href}
+                meta={meta}
+                imageUrl={imageUrl}
+                initials={null}
+                FollowButton={followButton}
+              />
+            </TouchableWithoutFeedback>
+          </Flex>
+        )}
+      </>
+    )
+  }
+}
+
+export const ContextCardFragmentContainer = createFragmentContainer(ContextCard, {
+  artwork: graphql`
+    fragment ContextCard_artwork on Artwork {
+      id
+      gravityID
+      internalID
+      context {
+        __typename
+        ... on ArtworkContextAuction {
+          id
+          name
+          href
+          formattedStartDateTime
+          cover_image {
+            url
+          }
+        }
+      }
+      shows(size: 1) {
```

Comment:
Is there a reason we're not spreading on the types for `show` and `fair`? (`... on ArtworkContextFair` and `... on ArtworkContextPartnerShow`)?

Reply:
The reason we're not for Show is the context doesn't have a `ArtworkContextPartnerShow` type yet. Can add it for `Fair`.

Reply:
Curious why we need to spread it in the context though if we're already querying for the fair directly on the Artwork.

Reply:
Gotcha on the `show`.

The way it's written right now you will _always_ be querying for `fair` on the artwork, but by spreading on the union type, you'll only fetch for the resource when it's part of the context that's returned.

Reply:
Cool just updated!